File: Python/BOJ/15654.py
import sys

n, m = map(int, sys.stdin.readline().split())

nums = list(map(int, sys.stdin.readline().split()))
nums.sort()
lst = []
def dfs():
    if len(lst) == m:
        print(' '.join(map(str, lst)))
        return
    for i in nums:
        if i not in lst:
            lst.append(i)
            dfs()
            lst.pop()

dfs()

File: C++/BOJ/15654.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int n, m;
vector<int> nums;
vector<int> lst;

void    dfs()
{
    if (lst.size() == m)
    {
        for (auto l : lst)
            cout<<l<<" ";
        cout<<"\n";
        return ;
    }
    for (auto num : nums)
    {
        if (find(lst.begin(), lst.end(), num) == lst.end())
        {
            lst.push_back(num);
            dfs();
            lst.pop_back();
        }   
    }
}

int main()
{
    
    int num;
    
    cin>>n>>m;

    for (int i = 0; i < n; i++)
    {
        cin>>num;
        nums.push_back(num);
    }
    sort(nums.begin(), nums.end());
    dfs();
}
